neon_data_models/models/api/http/brainforge.py

```diff
@@ -37,28 +44,173 @@ class LLMGetModelsHttpResponse(BaseModel):
 
 class LLMGetPersonasHttpRequest(BaseModel):
     model_id: str = Field(
-        description="Model ID (<name>@<version>) to get personas for")
+        description="Model ID (<name>@<version>) to get personas for"
+    )
 
 
 class LLMGetPersonasHttpResponse(BaseModel):
     personas: List[LLMPersona] = Field(
-        description="List of personas associated with the requested model.")
+        description="List of personas associated with the requested model."
+    )
 
 
 class LLMGetInferenceHttpRequest(LLMRequest):
     llm_name: str = Field(description="Model name to request")
     llm_revision: str = Field(description="Model revision to request")
     model: Optional[str] = Field(
-        default=None, 
-        description="Model ID (<name>@<version>) used for vLLM inference")
+        default=None,
+        description="Model ID (<name>@<version>) used for vLLM inference",
+    )
 
-    @model_validator(mode='after')
+    @model_validator(mode="after")
     def set_model_from_name_and_revision(self):
         if self.model is None:
             self.model = f"{self.llm_name}@{self.llm_revision}"
         return self
 
-__all__ = [LLMGetModelsHttpResponse.__name__,
-           LLMGetPersonasHttpRequest.__name__,
-           LLMGetPersonasHttpResponse.__name__,
-           LLMGetInferenceHttpRequest.__name__]
+
+class OpenAiCompletionRequest(BaseModel):
+    model: str = Field(description="Model ID (generally <name>@<version>)")
+    messages: List[Dict[Literal["role", "content"], str]] = Field(
+        description="List of messages to send to the model for completion, "
+        "including `system` and other context messages."
+    )
+    max_tokens: Optional[int] = Field(
+        default=512, description="Maximum number of tokens to generate"
+    )
+    temperature: Optional[float] = Field(
+        default=0.0, description="Temperature of generation"
+    )
+    stream: Optional[bool] = Field(
+        default=False, description="Not Implemented."
+    )
+    persona: Optional[LLMPersona] = Field(default=None)
```

Comment:
Do we also want to add `extra_body`?
Like [here](https://github.com/NeonGeckoCom/brainforge-webapp/blob/d92ce631791b1a8144064a3d96dcdef1ac1d9f7e/brainforge_webapp/app.py#L327)